### PR TITLE
Deprecate set_conf_file

### PR DESCRIPTION
**Pull request by SebastianBoe**
In-tree, all usage of 'set_conf_file' is implementing the same
pattern (with some inconsequential differences).

To reduce copy-paste and get closer to dropping support for
'set_conf_file', we include this pattern into the built-in set of
patterns.

'set_conf_file' is causing issues for future extensions,
e.g. multi-image support, and is also the source of a lot of
duplicated code.

Note support for 'set_conf_file' is not dropped, it is only marked as
deprecated.

### CODE REVIEW

**Comment by SebastianBoe** [comment]
@pfalcon : FYI

**Review by vanwinkeljan — APPROVED** [review]

File: samples/net/sockets/echo/CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,9 @@
...
 
 include($ENV{ZEPHYR_BASE}/cmake/app/boilerplate.cmake NO_POLICY_SCOPE)
 project(sockets_echo)
 
 FILE(GLOB app_sources src/*.c)
 target_sources(app PRIVATE ${app_sources})
 
-include($ENV{ZEPHYR_BASE}/samples/net/common/common.cmake)
\ No newline at end of file
+include($ENV{ZEPHYR_BASE}/samples/net/common/common.cmake)
```
Comment (jukkar) [inline samples/net/sockets/echo/CMakeLists.txt]: Unrelated change

File: tests/drivers/i2c/i2c_slave_api/boards/nucleo_f091rc.conf [hunk 2]
```diff
@@ -2,3 +2,4 @@ CONFIG_I2C_STM32_V2=y
 CONFIG_I2C_STM32_INTERRUPT=y
 CONFIG_I2C_1=y
 CONFIG_I2C_2=y
+CONFIG_I2C_VIRTUAL=n
```
Comment (jukkar) [inline tests/drivers/i2c/i2c_slave_api/boards/nucleo_f091rc.conf]: unrelated change

File: tests/drivers/i2c/i2c_slave_api/boards/stm32f072b_disco.conf [hunk 3]
```diff
@@ -2,3 +2,4 @@ CONFIG_I2C_STM32_V2=y
 CONFIG_I2C_STM32_INTERRUPT=y
 CONFIG_I2C_1=y
 CONFIG_I2C_2=y
+CONFIG_I2C_VIRTUAL=n
```
Comment (jukkar) [inline tests/drivers/i2c/i2c_slave_api/boards/stm32f072b_disco.conf]: unrelated change

**Review by jukkar — CHANGES_REQUESTED** [review]
Indeed, it is just useless to replicate the same code all over the place. So +1 for the change. There are some unrelated changes introduced in this PR, those should be removed and possibly submitted in a separate PR/commit.

File: samples/net/sockets/echo/CMakeLists.txt (hunk 1 above)
Comment (SebastianBoe) [reply samples/net/sockets/echo/CMakeLists.txt]: True, but, we should have a newline at the end of the file.

I can create another commit for this, but I would prefer to leave it as-is.

Comment (jukkar) [reply samples/net/sockets/echo/CMakeLists.txt]: Adding a new commit into this set should not be a big issue. Don't you need to remove the extra CONFIG_I2C_VIRTUAL settings anyway?

File: tests/drivers/i2c/i2c_slave_api/boards/nucleo_f091rc.conf (hunk 2 above)
Comment (SebastianBoe) [reply tests/drivers/i2c/i2c_slave_api/boards/nucleo_f091rc.conf]: Actually, it is related.

Unlike all other samples, i2c_slave_api had two different files for the "common" configuration.

One that was used when no board configuration was needed (prj_virtual.conf), and another that is combined with board config (prj_base.conf).

This pattern is not supported by boilerplate.cmake, so this sample was changed to only have one "common" config file (prj.conf). Doing so required modifying the board configuration files to disable some of the "common" configuration from prj_base.conf.

Comment (jukkar) [reply tests/drivers/i2c/i2c_slave_api/boards/nucleo_f091rc.conf]: Ok.

**Review by jukkar — APPROVED** [review]

**Review by dbkinder — APPROVED** [review]
+1 for doc changes

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nashif — APPROVED** [review]
